src/visor/visor.go

```diff
@@ -380,21 +390,53 @@ func (vs *Visor) CreateAndExecuteBlock() (coin.SignedBlock, error) {
 	return sb, err
 }
 
+// CreateBlock creates a Block from specified set of transactions according to set of determinstic rules.
+func (vs *Visor) CreateBlock(txns coin.Transactions, when uint64) (coin.Block, error) {
+	var sb coin.Block
+
+	err := vs.db.Update("CreateBlock", func(tx *dbutil.Tx) error {
+		var err error
+		if sb, err = vs.createBlockFromTxns(tx, txns, when); err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	return sb, err
+}
+
+// VerifyBlock verifies specified block against local copy of blockchain.
+// Signature is not verified.
+func (vs *Visor) VerifyBlock(b coin.SignedBlock) error {
+	return vs.db.View("VeirfyBlock", func(tx *dbutil.Tx) error {
+		return vs.blockchain.VerifyBlock(tx, &b)
+	})
+}
+
 // ExecuteSignedBlock adds a block to the blockchain, or returns error.
-// Blocks must be executed in sequence, and be signed by a block publisher node
+// Blocks must be executed in sequence, and be signed by a block publisher node.
 func (vs *Visor) ExecuteSignedBlock(b coin.SignedBlock) error {
+	if err := b.VerifySignature(vs.Config.BlockchainPubkey); err != nil {
+		return err
+	}
+
+	return vs.db.Update("ExecuteSignedBlock", func(tx *dbutil.Tx) error {
+		return vs.executeSignedBlock(tx, b)
+	})
+}
+
+// ExecuteSignedBlockUnsafe adds block to the blockchain, or returns error.
```

Comment:
Explain what this is for?

Reply:
That is needed for applications, which Are going to perform block signing and block signature verification on their own.